### PR TITLE
Disable ALTHOLD on user motor stop request

### PR DESCRIPTION
**Pull request by shellixyz**
Before if the user puts the throttle stick bellow `mincheck` when ALTHOLD (all nav states using NAV_CTL_ALT) is enabled and the MOTOR_STOP feature is enabled the motor is stopped while ALTHOLD still tried to maintain altitude. This can quickly lead to a stall.

With this patch ALTHOLD is paused when the throttle is bellow `mincheck` (motor stopped if the MOTOR_STOP feature is enabled or at minthrottle otherwise) and resumed when the user throttles up again.

It allows to start gliding just by putting the throttle stick to 0 and start maintaining altitude again by increasing throttle above `mincheck`

### CODE REVIEW

File: src/main/fc/runtime_config.h
Comment (digitalentity) [inline src/main/fc/runtime_config.h]: Wouldn't it be easier to allow mixer expose status of motor?
I.e. `getMotorStatus` with possible values of `MOTOR_RUNNING`, `MOTOR_STOPPED_USER`, `MOTOR_STOPPED_AUTO`?

Comment (shellixyz) [reply src/main/fc/runtime_config.h]: Good idea. I added a new `getMotorStatus` function in mixer.c and removed the state flag `USER_MOTOR_STOP`

File: src/main/navigation/navigation.c
Comment (digitalentity) [inline src/main/navigation/navigation.c]: I believe when we exit the "pause" we should reset the altitude setpoint - hence `setDesiredPosition` should be called here, before switching to "in-progress" state

File: src/main/navigation/navigation_fixedwing.c
Comment (digitalentity) [inline src/main/navigation/navigation_fixedwing.c]: Why checking for motor-stop here? When ALTHOLD is paused NAV_CTL_ALT is not set anyway.

Comment (shellixyz) [reply src/main/navigation/navigation_fixedwing.c]: True ! It is a reminiscence of old code. Fixed

**Review by digitalentity — APPROVED** [review]
Look ok now. Some testing required before we can merge this.

**Comment by digitalentity** [comment]
Once we have full featured TECS with airspeed we'll have to revisit this - TECS will be able to pitch down to maintain airspeed so it will have to be running at all times.

**Comment by shellixyz** [comment]
Please don't merge this yet I need to change the way it is working to be able to do the same for the upcoming 3D CRUISE mode.

**Comment by shellixyz** [comment]
I modified this PR. I think it is much better now, more polyvalent. No need to modify the ALTHOLD states and it will be working with 3D cruise mode without changing anything to it either.

I also took the opportunity to simplify/deduplicate code in `applyFixedWingNavigationController()`

Tested and now ready to merge.

**Review by digitalentity — APPROVED** [review]
LGTM

**Pull request merged**
